drivers/ethernet/eth_lan865x.c

```diff
@@ -437,31 +414,18 @@ static int lan865x_init(const struct device *dev)
 		return ret;
 	}
 
+	oa_tc6_init(tc6);
+
 	return lan865x_gpio_reset(dev);
 }
 
 static int lan865x_port_send(const struct device *dev, struct net_pkt *pkt)
 {
 	struct lan865x_data *ctx = dev->data;
 	struct oa_tc6 *tc6 = ctx->tc6;
-	int ret;
 
-	k_sem_take(&ctx->tx_rx_sem, K_FOREVER);
```

Comment:
IMHO this patch (i.e. "drivers: ethernet: update OA TC6 lib & lan865x driver for max performance") shall be:
1. Removed from this PR, as we are now focusing on PHY support for lan865x and not rewriting this driver to get better performance. For the full-duplex driver we shall have separate PR.
2. As it was pointed out - this driver increases considerably the memory footprint (and we don't know yet about its stability). Please be aware that for some use cases the half-duplex performance is just good enough.
3. I've written down some summary regarding testing performance of the driver:
https://github.com/zephyrproject-rtos/zephyr/issues/74393
Maybe modifying the code to use single transfer for header/footer+payload (point 1. of above issue) would increase the performance to acceptable level?
4. I was trying to investigate the possibility to write full-duplex OA TC6 driver (https://github.com/lmajewski/zephyr/commit/fb3adfee45c796ce2078458d089c775b2f05296d), but from my measurements it turned out that by using other tricks one can improve the performance.

Reply:
> This is needed because the T1S n/w is capable to transfer 10Mbps data and if the platform also is capable to support in terms of memory then it should be able to support max of 9.4Mbps

+1

> One idea could be giving all the memory size relevant params as user configurable based on their platform need.

+1

Probably - we shall have `CONFIG_OA_TC6_DMA` or `CONFIG_OA_TC6_CONTIGUOUS_MEMORY` switch - which would distinct the case when contiguous memory (i.e. NOT fragments) is required (for me large memory allocations are problematic).

I'm also wondering if HALF / FULL duplex shall be optional - this all depends on the resources required and if we DO require the DMA for proper full duplex operation (on my platform the `CONFIG_SPI_STM32_DMA` is not set). However, I would love to have the full duplex driver version even when DMA is not enabled (or I will just investigate if DMA can be safely used in my use case). If NOT we can keep HALF duplex driver as an option.